druid/src/window.rs

```diff
@@ -273,8 +273,17 @@ impl<T: Data> Window<T> {
     }
 
     pub(crate) fn invalidate_and_finalize(&mut self) {
-        if self.root.state().needs_inval {
+        // Currently, if we need to re-layout then we invalidate the whole window.
+        // We could be smarter about this (only invalidating the part of the widget tree
+        // that changed its layout), but I think this would require doing the layout
+        // before the paint pass, so that the system knows which part got invalidated.
```

Comment:
we do always do layout (if needed) before paint? This interaction is subtle; a layout invalidation implies a paint invalidation, but we could be smarter about stopping invalidation when old and new layouts converge? That's definitely out of scope for this PR.

To clarify: when any widget has its layout invalidated we should be repainting the whole view.

Reply:
I agree that this it's definitely out of scope, and I guess a github issue would be better than a comment that no one will read.
Anyway, this was triggered by the calc example, where every time you press a number the layout gets invalidated, even though only the calculator's "screen" changes its layout.